lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2315,12 +2338,16 @@ class HSADevice final : public KalmarDevice
         // This defers expensive queue deallocation if an hccQueue that holds an hwQueue is destroyed -
         // keep the hwqueue around until the number of hccQueues drops below the number of hwQueues
         // we have already allocated.
-        auto rqSize = rocrQueues.size();
+        // TODO: Do we want to track HCC queues independently for each priority?
+        // Or maybe modify queue stealing to steal unused HSA queue from other priority pools &
+        // change priority as required?
+        auto rqSize = rocrQueues[0].size()+rocrQueues[1].size()+rocrQueues[2].size();
```

Comment:
I think this heuristic needs to be refined since it doesn't seem right that we only compare the global sizes and then decide on whether to remove a queue for a particular priority

Reply:
How do you want this to be handled? As mentioned in the TODO, ```hccSize``` currently tracks number of av instances across all priorities. Do we want to track it independently for each priority?
Is it fine to address this in a follow up patch?

Reply:
ok, I'm fine with a follow up PR